### DISCUSSION TITLE
intellisenseMode understanding

### DISCUSSION THREAD

**Discussion by boocs** [microsoft/vscode-cpptools#9747] [Q&A] [answered]
There are 3 parts to an intellisenseMode. I'm trying to understand the 3rd part.

The 1st part is not the OS you're using but what your project is targeting: Found here https://visualstudiomagazine.com/articles/2021/02/11/vscode-cpp-update.aspx

The 2nd part is easy and is the compiler you're using.

So what is the 3rd part? Let's say you have your intellisenseMode set to "macos-clang-x64"

So this means you're targeting macos, using clang, and
your project is also targeting x64
or
you're using a x64 clang build?

**Comment by bobbrow** [comment, marked as answer]
It means you are targeting x64.  So for example, `sizeof(int*)` should show that the pointer size is 8 instead of 4.
